akka-sample-distributed-data-scala/src/main/scala/sample/distributeddata/ReplicatedMetrics.scala

```diff
@@ -36,80 +41,104 @@ object ReplicatedMetrics {
 
   def nodeKey(address: Address): String = address.host.get + ":" + address.port.get
 
-}
-
-class ReplicatedMetrics(measureInterval: FiniteDuration, cleanupInterval: FiniteDuration)
-  extends Actor with ActorLogging {
-  import akka.cluster.ddata.Replicator._
-  import ReplicatedMetrics._
-
-  val replicator = DistributedData(context.system).replicator
-  implicit val cluster = Cluster(context.system)
-  val node = nodeKey(cluster.selfAddress)
-
-  val tickTask = context.system.scheduler.schedule(measureInterval, measureInterval,
-    self, Tick)(context.dispatcher)
-  val cleanupTask = context.system.scheduler.schedule(cleanupInterval, cleanupInterval,
-    self, Cleanup)(context.dispatcher)
-  val memoryMBean: MemoryMXBean = ManagementFactory.getMemoryMXBean
-
-  val UsedHeapKey = LWWMapKey[String, Long]("usedHeap")
-  val MaxHeapKey = LWWMapKey[String, Long]("maxHeap")
-
-  replicator ! Subscribe(UsedHeapKey, self)
-  replicator ! Subscribe(MaxHeapKey, self)
-
-  cluster.subscribe(self, InitialStateAsEvents, classOf[MemberUp], classOf[MemberRemoved])
-
-  override def postStop(): Unit = {
-    tickTask.cancel()
-    cluster.unsubscribe(self)
-    super.postStop()
-  }
-
-  var maxHeap = Map.empty[String, Long]
-  var nodesInCluster = Set.empty[String]
-
-  def receive = {
-    case Tick =>
-      val heap = memoryMBean.getHeapMemoryUsage
-      val used = heap.getUsed
-      val max = heap.getMax
-      replicator ! Update(UsedHeapKey, LWWMap.empty[String, Long], WriteLocal)(_ + (node -> used))
-      replicator ! Update(MaxHeapKey, LWWMap.empty[String, Long], WriteLocal) { data =>
-        data.get(node) match {
-          case Some(`max`) => data // unchanged
-          case _           => data + (node -> max)
+  def apply[A <: Nothing]: Behavior[A] = apply[A](1.second, 1.minute)
```

Comment:
I'd prefer making `Command` public and then have the public signature `Behavior[Command]´ with no (current) public messages over the `A <: Nothing` trixery.